File: 0747-min-cost-climbing-stairs/0747-min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int a =0, b=0;
        int n=cost.size();
        for(int i=2;i<=n;i++){
            int t= a;
            a = min(a+cost[i-1], b+cost[i-2]);
            b=t;
        }
        return a;
    }
};

File: 0747-min-cost-climbing-stairs/0747-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int a =0,b=0;
        for(int i=2;i<=cost.length;i++){
            int t = a;
            a = Math.min(a+cost[i-1],b+cost[i-2]);
            b=t;
        }
        return a;
    }
}